src/smoke.hh

```diff
@@ -0,0 +1,19 @@
+#include <nanogui/nanogui.h>
+
+#include "emittor.hh"
+
+using namespace nanogui;
+using namespace std;
+
+class Smoke
+{
+
+public:
+    Smoke();
+    ~Smoke(){};
+
+    vector<Particle> particles;
```

Comment:
Maybe we should use list here so that we can update (i.e., create / remove particles) in-place. Copy this large vector every time is expensive.